src/stream/src/common/table/state_table.rs

```diff
@@ -806,13 +943,32 @@ where
         } else {
             Some(self.pk_serde.prefix(1))
         };
-        let watermark_suffix = watermark.map(|watermark| {
+
+        let should_clean_watermark = match watermark {
+            Some(ref watermark) => {
+                if USE_WATERMARK_CACHE && self.watermark_cache.is_synced() {
+                    if let Some(key) = self.watermark_cache.lowest_key() {
+                        watermark.as_scalar_ref_impl().default_cmp(&key).is_ge()
+                    } else {
+                        // If empty and synced, means no rows in table.
+                        self.watermark_cache.is_empty()
```

Comment:
If there is no rows in table. Why set `should_clean_watermark` ?
Is it a typo?

Reply:
Yes, this should be removed.